src/Orleans.TestingHost/TestClusterHostFactory.cs

```diff
@@ -59,13 +59,10 @@ public static ISiloHost CreateSiloHost(string hostName, IEnumerable<IConfigurati
                 TryConfigureTestClusterMembership(context, services);
                 TryConfigureFileLogging(configuration, services, siloName);
 
-                // TODO: make SiloHostBuilder work when not using the legacy configuration, similar to what we did with ClientBuilder.
-                // All The important information has been migrated to strongly typed options (everything should be migrated, but the minimum required set is already there).
-                var clusterConfiguration = GetOrCreateClusterConfiguration(services);
```

Comment:
if all the defaults are indeed set in options now, then this step can be removed and tests should all be passing. Otherwise this PR may depends on #3949 or we can finish #3949 in this PR.